exporter/datadogexporter/utils/cache/cache.go

```diff
@@ -0,0 +1,48 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cache
+
+import (
+	"time"
+
+	gocache "github.com/patrickmn/go-cache"
+)
+
+const (
+	CanonicalHostnameKey = "canonical_hostname"
+	SystemHostInfoKey    = "system_hostinfo"
+)
+
+var generalCache = gocache.New(20*time.Minute, 10*time.Minute)
```

Comment:
I feel like having a global backing store may be fine, but I think it would be nice to at least pass it or a wrapper into each function or type that needs it. That way you make the dependency explicit. That would also mean that the functions below as written could be deleted.

Reply:
The functions below are mostly there for ease at refactoring if we want to change the cache, but I think it is simple enough that I can get rid of them, yeah, I will change it to expose and pass the `generalCache`.

Reply:
I removed the functions and exposed the cache!